### PR TITLE
feat: created basic section and their forms

### PR DESCRIPTION
**Pull request by sivangbagri**
## Description

Fixes #(issue)

## Checklist

- [x] My code follows the [contributing guidelines][contributing] of this project.
- [x] I am aware that all my commits will be squashed into a single commit, using the PR title as the commit message.
- [ ] I have performed a self-review of my own code.
- [x] I have commented my code in hard-to-understand areas.
- [ ] I have updated the user-facing documentation to describe any new or changed behavior.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have not reduced the existing code coverage.

## Comments

[contributing]: CONTRIBUTING.md

## Summary by Sourcery

Add a new 'ECCCLientRoCrateAbout' component with tabbed forms for managing metadata sections, and provide demo HTML files for component demonstration.

New Features:
- Introduce a new web component 'ECCCLientRoCrateAbout' using LitElement, which includes forms for 'About', 'Related People, Orgs & Works', and 'Structure' sections.

Enhancements:
- Add a tabbed interface to switch between different sections ('About', 'Related People, Orgs & Works', and 'Structure') within the 'ECCCLientRoCrateAbout' component.

Documentation:
- Add demo HTML files to showcase the usage of the new 'ECCCLientRoCrateAbout' component, including a basic setup and integration example.

### CODE REVIEW

**Comment by sivangbagri** [comment]
@salihu @anuragxxd  Kindly review. Ignore naming issues , I will fix it

**Comment by SalihuDickson** [comment]
hi @sivangbagri, I'm having a bit of trouble reviewing your code, can you please create an issue that goes into detail about the goals of this project and how you are going about achieving said goals, also include important links like the schema for the version of ro-crate you're working with. If there is one already please link it in the PR description.

**Comment by sivangbagri** [comment]
@SalihuDickson Hii , check this out : https://github.com/elixir-cloud-aai/cloud-components/issues/356 . If anything else is needed please let me know

File: packages/ecc-client-elixir-ro-crate/src/components/about/about.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    {
+      key: "author",
+      label: "Author",
+      type: "text",
+      fieldOptions: {
+        default: "Choose a Type",
+      },
+    },
+    {
+      key: "pulisher",
```
[the remark this answers is not in the document]
Comment (anuragxxd) [reply packages/ecc-client-elixir-ro-crate/src/components/about/about.ts:98]: Incorrect Spelling!

File: packages/ecc-client-elixir-ro-crate/src/components/about/about.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
+import { LitElement, html } from "lit";
+import { state } from "lit/decorators.js";
+// import EccUtilDesignForm from "@elixir-cloud/design/dist/react/form/index";
+import "@elixir-cloud/design/dist/components/form/index.js";
+import { Field } from "@elixir-cloud/design/dist/components/form/form";
+
+export default class ECCCLientRoCrateAbout extends LitElement {
```
[the remark this answers is not in the document]
Comment (anuragxxd) [reply packages/ecc-client-elixir-ro-crate/src/components/about/about.ts:7]: This is also true!

**Review by anuragxxd — CHANGES_REQUESTED** [review]
Some small changes otherwise lgtm

**Comment by sivangbagri** [comment]
@anuragxxd new commit has required changes along with improved form structure of about and hasPart sections.

**Review by SalihuDickson — COMMENTED** [review]
Just a few things:

1. The Field is `@type`, not `Type`
2. The `@id` field should be required
3. What is the "about dataset" section for? Are the fields above that not for the root data entity (the dataset) entity.

Some discrepancies between from the latest official [Ro-Crate sepcification](https://www.researchobject.org/ro-crate/):
1.  The `@type` field is not needed as both the metadata and the dataset entities have static types, you only need the `@type` field when you are creating a new entity.
2. Name, description and date published should be required, there is also a required license entity that you need to add fields for.

Great work so far @sivangbagri, please take a look at the specification for the latest version and make necessary adjustments. And if you have any questions please feel free to reach out.

**Comment by SalihuDickson** [comment]
The fields you've added for the Author, Publisher and Funder fields won't really work, based on the minimum requirements for the Ro-Crates.

The author object is would be an instance of a [people entitiy](https://www.researchobject.org/ro-crate/specification/1.1/contextual-entities.html#people).
The Publisher object would be an instance of an [organization entity](https://www.researchobject.org/ro-crate/specification/1.1/contextual-entities.html#publisher) or a people entity 
The Funder object can also be an Organization or a people entity

perhaps the "Related People, Orgs & Works" page can contain all contextual entities, but we would need a better design to make that clear, as the current structure implies several disjointed entities.

This is a lot at this point, my suggestion would be that you convert this to a draft and create a new PR solely adding fields for the metadata entity, another for the dataset entity and then we can address these other objects and entities in subsequent PRs. 

Please let me know if there's anything i need to elaborate on.

**Comment by SalihuDickson** [comment]
I was in the process of trying to revert the ro-crate changes and merge the build fixes from earlier but i just realized those have already been implemented on main. I think you can go ahead and close this PR.

**Pull request closed without merging**